NeuroID/Utilities/NIDConfigService.swift

```diff
@@ -6,56 +6,95 @@ import Alamofire
+        self.networkService = networkService
+    }
     
-    init(completion: @escaping (Bool) -> Void) {
-        if NeuroID.clientKey == nil || NeuroID.clientKey == "" {
-            NIDLog.e("Missing Client Key. Config Service not started")
-            completion(true)
+    func retrieveConfig(completion: @escaping () -> Void) {
+        if !NeuroID.verifyClientKeyExists() {
+            cacheSetWithRemote = false
+            completion()
             return
         }
-                     
-        let config_url = NIDConfigService.nidURL + NeuroID.clientKey! + ".json"
-        AF.request(config_url, method: .get).responseDecodable(of: ResponseData.self) { response in
+        
+        let config_url = NIDConfigService.NID_CONFIG_URL + NeuroID.clientKey! + ".json"
+        networkService.getRequest(
+            url: URL(string: config_url)!,
+            responseDecodableType: ConfigResponseData.self
+        ) { response in
             switch response.result {
             case .success(let responseData):
                 self.setCache(responseData)
-                NIDLog.d("Retrieved config log")
-                NIDConfigService.cacheSetWithRemote = true
-                completion(true)
+                NIDLog.d("Retrieved remote config")
+                
+                self.cacheSetWithRemote = true
+                self.cacheCreationTime = .init()
+                completion()
             case .failure(let error):
                 NIDLog.e("Failed to retrieve NID Config \(error)")
-                NIDConfigService.nidConfigCache = ResponseData()
-                completion(true)
+                self.configCache = ConfigResponseData()
+                self.cacheSetWithRemote = false
+                completion()
             }
         }
     }
     
-    func setCache(_ responseData: ResponseData) {
-        NIDConfigService.nidConfigCache = responseData
+    func setCache(_ newCache: ConfigResponseData) {
+        configCache = newCache
+    }
+    
+    func expiredCache() -> Bool {
+        if !cacheSetWithRemote {
+            return true
+        }
+        // 5 min is the default, can be updated later
+        let TTLTime = Calendar.current.date(byAdding: .minute, value: -5, to: Date())!
+        
+        return cacheCreationTime < TTLTime
+    }
+    
+    func retrieveOrRefreshCache(completion: @escaping () -> Void) {
+        if expiredCache() {
+            retrieveConfig {
+                completion()
+            }
+        } else {
+            completion()
+        }
+    }
+    
+    func updateConfigOptions(siteID: String? = nil, completion: @escaping () -> Void) {
+        // check cache time
+        retrieveOrRefreshCache {
+            // retrieve the site config for the site
+            
+            // if siteID == config.siteID - then use top level value
+            // if siteID == nil then assume parent site
+            if siteID == nil || siteID ?? "" == self.configCache.siteID ?? "noID" {
```

Comment:
minor nit: `siteID == nil || siteID ?? "" == self.configCache.siteID ?? "noID"` could be written as a function `isCollectionSite(siteId)` to make this more readable